### PR TITLE
Fixed various typos across the documentation.

### PR DESCRIPTION
**Pull request by Xenoparrot**
Most typos fixed are related to the word "an". Though there are a few unrelated typos.

### CODE REVIEW

File: doc/classes/SyntaxHighlighter.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	</brief_description>
 	<description>
 		Base syntax highlighter resource all syntax highlighters extend from, provides syntax highlighting data to [TextEdit].
-		The associated [TextEdit] node will call into the [SyntaxHighlighter] on a as needed basis.
+		The associated [TextEdit] node will call into the [SyntaxHighlighter] on an as needed basis.
```
Comment (MewPurPur) [inline doc/classes/SyntaxHighlighter.xml:8]: "as-needed"

**Review by MewPurPur — APPROVED** [review]
The changes are all correct, I suggested one further change on one of the changed lines.

File: doc/classes/SyntaxHighlighter.xml (hunk 1 above)
Comment (Xenoparrot) [reply doc/classes/SyntaxHighlighter.xml]: Nice catch, thanks

File: doc/classes/OS.xml [hunk 2]
```diff
@@ -426,7 +426,7 @@
 			<description>
 				Returns an array of the system substitute font file paths, which are similar to the font with [param font_name] and style for the specified text, locale and script. Returns empty array if no matching fonts found.
 				The following aliases can be used to request default fonts: "sans-serif", "serif", "monospace", "cursive", and "fantasy".
-				[b]Note:[/b] Depending on OS, it's not guaranteed that any of the returned fonts is suitable for rendering specified text. Fonts should be loaded and checked in the order they are returned, and the first suitable one used.
+				[b]Note:[/b] Depending on OS, it's not guaranteed that any of the returned fonts are suitable for rendering specified text. Fonts should be loaded and checked in the order they are returned, and the first suitable one used.
```
Comment (AThousandShips) [inline doc/classes/OS.xml:429]: I'd suggest "will be suitable", to me it flows better and the tense fits better

Comment (AThousandShips) [reply doc/classes/OS.xml:429]: And from what I can find both "is" and "are" are valid here, and carries slightly different meaning, as this talks about that it doesn't guarantee the validity of any of the fonts I would argue "is" to be more correct, but neither is necessarily incorrect

Comment (Xenoparrot) [reply doc/classes/OS.xml:429]: Thanks for the feedback, I appreciate it. I wholeheartedly agree with the suggestion, I'll update it soon. I was thinking it myself but decided to keep it simple. Which leads to your next point...

As far as "is" vs "are": whoops. I didn't realize that "any of" is different than "any". Nice catch. I can't believe I created a typo trying to fix one. I'll remember for future reference.

Comment (Xenoparrot) [reply doc/classes/OS.xml]: Yeah I agree. Neither is necessarily wrong, it's just that the singular is considered more appropriate in a formal context (according to the below link) and thus I was wrong to change it.

https://english.stackexchange.com/questions/475028/is-any-of-them-used-with-a-singular-or-plural-verb

Regardless, saying "will be" sounds better anyway and so I've updated it.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for these typo fixes, and congrats on your first merged contribution to Godot! 🎉

**Comment by akien-mga** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**